### PR TITLE
Liqun/container2

### PR DESCRIPTION
**Pull request by liqul**
1. switch to `container` mode as the default execution mode
2. change container name and allow pulling from docker hub
3. change container user to `taskweaver` instead of `root`
4. refactor docker entry to single kernel process

### CODE REVIEW

File: taskweaver/ces/environment.py [hunk 1]
```diff
@@ -230,31 +218,29 @@ def start_session(
...
-            os.makedirs(ces_session_dir, exist_ok=True)
-            cwd = cwd if cwd is not None else os.path.join(session.session_dir, "cwd")
-            os.makedirs(cwd, exist_ok=True)
             connection_file = self._get_connection_file(session_id, new_kernel_id)
             new_port_start = self.port_start_inside_container
             kernel_env = {
-                "TASKWEAVER_ENV_ID": self.id,
-                "TASKWEAVER_SESSION_ID": session_id,
-                "TASKWEAVER_KERNEL_ID": new_kernel_id,
-                "TASKWEAVER_ENV_DIR": "/app",
```
Comment (Jack-Q) [inline taskweaver/ces/environment.py:248]: These few variable are still referred in the executor script even in container mode and might could have them reserved as well?

File: taskweaver/ces/environment.py [hunk 2]
```diff
@@ -230,31 +218,29 @@ def start_session(
 
             self._cmd_session_init(session)
             session.kernel_status = "ready"
+        elif self.mode == EnvMode.Container:
+            if platform.system() != "Windows":
+                # change the permission of the session directory and its subdirectories
+                os.chmod(session.session_dir, 0o755)
+                os.chmod(ces_session_dir, 0o755)
+                os.chmod(cwd, 0o755)
```
Comment (Jack-Q) [inline taskweaver/ces/environment.py:226]: just curious why we need to explicitly configure those folders to 755? (also session dir might not needed if it has been removed below from mounting)

File: taskweaver/module/execution_service.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ def _configure(self) -> None:
         )
         self.kernel_mode = self._get_str(
             "kernel_mode",
-            "local",
+            "container",
         )
         if self.kernel_mode == "local":
             print(
```
[the remark this answers is not in the document]
Comment (liqul) [reply taskweaver/module/execution_service.py:23]: Thought about this. But one key problem is to separate the main process (containing the configs) with the execution process into different envs. So, I think this warning is still valid.

File: taskweaver/ces/manager/sub_proc.py
[the remark this answers is not in the document]
Comment (liqul) [reply taskweaver/ces/manager/sub_proc.py:1]: The problem now is that all implementations in environment.py and the manager and client are just an interface of that. We really need to consider how to reorg this part if we need to have other managers.

File: taskweaver/ces/environment.py (hunk 2 above)
Comment (liqul) [reply taskweaver/ces/environment.py:226]: session is not needed. for others, I didn't test them but have to make them accessable by taskweaver inside the container in Linux env.

File: taskweaver/ces/environment.py [hunk 4]
```diff
@@ -164,10 +144,21 @@ def __init__(
             except docker.errors.DockerException as e:
                 raise docker.errors.DockerException(f"Failed to connect to Docker daemon: {e}. ")
 
+            self.image_name = "taskweavercontainers/taskweaver-executor"
```
[the remark this answers is not in the document]
Comment (liqul) [reply taskweaver/ces/environment.py:147]: good point, will do

File: scripts/build.sh [hunk 5]
```diff
@@ -4,7 +4,7 @@
 scriptDirectory="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 echo "The script directory is: $scriptDirectory"
 
-imageName="taskweaver/executor"
+imageName="taskweavercontainers/taskweaver-executor"
 taskweaverPath="$scriptDirectory/../taskweaver"
 dockerfilePath="$scriptDirectory/../ces_container/Dockerfile"
 contextPath="$scriptDirectory/../"
```
[the remark this answers is not in the document]
Comment (liqul) [reply scripts/build.sh:10]: not a big issue now ~

File: taskweaver/ces/environment.py (hunk 1 above)
Comment (liqul) [reply taskweaver/ces/environment.py:248]: will keep some

File: taskweaver/ces/kernel/docker_launcher.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    app.language = "python"
+
+    # get config from env
+    app.connection_file = connection_file
+    app.shell_port = port_start
+    app.iopub_port = port_start + 1
+    app.stdin_port = port_start + 2
+    app.hb_port = port_start + 3
+    app.control_port = port_start + 4
+    app.ip = ip
```
[the remark this answers is not in the document]
Comment (liqul) [reply taskweaver/ces/kernel/docker_launcher.py:54]: will do

**Review by Jack-Q — APPROVED** [review]

**Pull request merged**